src/server/main.js

```diff
@@ -63,9 +70,22 @@ io.on('connect', (socket) => {
    returns JWT
    */
   socket.on('login', async (user) => {
-    const login = await authentication.login(user);
+    const token = await authentication.login(user);
+
+    if (token.token) {
+      socketUser = authentication.verifyToken(token);
+    }
+
+    socket.emit('login', token);
+  });
+
+  socket.on('uploadMemeData', async (data) => {
+    if (socketUser === false) {
+      return socket.emit('uploadMemeData', 'cannot verify user');
+    }
+    const saveResult = await meme.saveMeme(data, socketUser);
```

Comment:
This event might need refactoring if we decide to send the image through socket io. If we upload to cloudinary from the client instead, then this won't need any changes.

Reply:
That is true, but I think uploading the file to cloudinary would be easier if done from the frontend.